numba/cpython/unicode.py

```diff
@@ -2470,6 +2453,219 @@ def boolean_str(b):
         return lambda b: "True" if b else "False"
 
 
+# WARNING: Integer in [-1<<63, 1<<63) is supported, which differs from cpython.
+# Global error messages for conversion (unicode -> integer)
+_INVALID_LITERAL_MSG = "invalid literal for int() with base"
```

Comment:
This exception string appears even when used in a case where no `base` is present, which is a little confusing. For example:
```python
@njit
def foo():
    return int('a') # there is no base given
```
It would be good to differentiate between the case of "no base" and "base given but not compatible".

Reply:
Same as above applies for a case like:
```python
@njit
def foo():
    return int('1', -2) # invalid base
```
would be good to have a more explicit message about what the problem is (i.e. base is out of supported range).